### DISCUSSION TITLE
ScriptReference/Object.DontDestroyOnLoad

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#360] [Notes]
# ScriptReference/Object.DontDestroyOnLoad

https://docs.unity3d.com/ScriptReference/Object.DontDestroyOnLoad.html

**Comment by docscomments** [comment]
**Comment by `@Baste-RainGames`:**  

There's a lot of information missing here, especially with regards to changing the parenting relation between objects.  
The rules for this method are these:  
- Calling DontDestroyOnLoad on an object that has a parent fails.  
- Calling DontDestroyOnLoad on a Component is equivalent to calling DontDestroyOnLoad on that Component's GameObject.  
- Calling DontDestroyOnLoad on an object marks that object and all it's children as DontDestroyOnLoad.  
- If you set object B as the parent of object A, object A will immediately and permanently get object B's DontDestroyOnLoad status.   
- Setting an object's parent to null (moving it to the root of the scene) doesn't change it's DontDestroyOnLoad status.

**Comment by jschiff** [comment]
DontDestroyOnLoad moves the object to a special scene called "DontDestroyOnLoad". This scene is special in that, when you call SceneManager.LoadScene in "Single" mode, it is the only scene which will not automatically be unloaded.
